pysb/simulator/templates/opencl.cl

```diff
@@ -0,0 +1,149 @@
+#include <pyopencl-random123/philox.cl>
+#define num_species {n_species}
+#define NPARAM {n_params}
+#define NREACT {n_reactions}
+#define NREACT_MIN_ONE NREACT-1
+
+typedef double output_t;
+typedef double4 output_vec_t;
+typedef philox2x32_ctr_t ctr_t;
+typedef philox2x32_key_t key_t;
+
+#define GET_RANDOM_NUM(gen) ( ((double) 2.3283064365386963e-10) * convert_double4(gen)+ ((double) 5.421010862427522e-20) * convert_double4(gen))
```

Comment:
Could you explain the meaning of these [magic numbers](https://en.wikipedia.org/wiki/Magic_number_(programming))? I can mostly see what this code is doing, but a little explanation about the above and the `gen_bits` function would be helpful.

Reply:
Fixed magic numbers and added citations for RNG